app/src/main/java/com/rob729/quiethours/Activity/MainActivity.kt

```diff
@@ -38,4 +44,15 @@ class MainActivity : AppCompatActivity() {
         val navController = myNavHostFragment.view?.let { Navigation.findNavController(it) }
         return navController!!.navigateUp() || super.onSupportNavigateUp()
     }
+
+    fun checkFirstRun() {
+        // Get current version code
```

Comment:
instead of this slightly complex implementation, you could have used a shared pref by the name "firstRun"/"firstBoot". On the first time opening the app, the value will be false as it will not be present in the database and after the first run set the value to true.
Also, try to shift this code to the splash screen and see if it works fine.
It will help in distributing the code among different files.

Reply:
@rob729 Thanks for your suggestion.
I have pushed the updated code reflecting upon the same.
Also, I have put checkFirstRun() in SplashScreen, it's working fine.